lib/backend/syncersv1/updateprocessors/workloadendpointprocessor.go

```diff
@@ -114,6 +114,11 @@ func convertWorkloadEndpointV2ToV1Value(val interface{}) (interface{}, error) {
 		})
 	}
 
+	// Felix does not have the concept of namespaces, so add the namespace as a Label.  We'll
```

Comment:
Hm, I wasn't expecting this code here, but I'm not sure what to make of it.

My expectation was that each plugin would write the namespace in.  This is nice because then they don't have to, but it also makes for an asymmetrical UX between defining a policy using certain label and being unable to query and see that label in, say `calicoctl`.

Reply:
If anything, this logic should go in the defaulting for WorkloadEndpoints so that whenever a WEP is written it gets this label on it (plus PodToWorkloadEndpoint for KDD).

That's a better approach I think.

Reply:
hmm not sure i agree about not writing in the namespace.

Otherwise the concept of namespace vs global resources is completely dependent on the behavior of the plugin - that doesn't make sense to me.

Eventually we should add the concept of namespace to felix, so we can have namespace selectors and then all this label manipulation for the namespace goes away - that's not the case if the orchestrator is trying to control that.

Reply:
I could possible buy that we return the namespace label in the policy definition - but that would be nothing to do with the orchestrator though - more just something in the client to make it clear to users what is going on.

Reply:
In fact I would go one further and say that we should be adding the Orchestrator label here too, rather than within the KDD code - since we know the orchestrator from the name/Spec.

And in the client Get/List code insert the same labels so that people can at least see labels we are "auto-inserting".

If it isn't added generically then the behavior of what the namespace implies is different depending on the orchestrator - but I think we probably want that behavior defined.